app/scripts/controllers/app-state-controller.ts

```diff
@@ -0,0 +1,867 @@
+import { AUTO_LOCK_TIMEOUT_ALARM } from '../../../shared/constants/alarms';
+import { isManifestV3 } from '../../../shared/modules/mv3.utils';
+// TODO: Remove restricted import
+// eslint-disable-next-line import/no-restricted-paths
+import { isBeta } from '../../../ui/helpers/utils/build-types';
+import {
+  ENVIRONMENT_TYPE_BACKGROUND,
+  POLLING_TOKEN_ENVIRONMENT_TYPES,
+  ORIGIN_METAMASK,
+} from '../../../shared/constants/app';
+import { DEFAULT_AUTO_LOCK_TIME_LIMIT } from '../../../shared/constants/preferences';
+import { LastInteractedConfirmationInfo } from '../../../shared/types/confirm';
+import { SecurityAlertResponse } from '../lib/ppom/types';
+import type {
+  Preferences,
+  PreferencesControllerGetStateAction,
+  PreferencesControllerStateChangeEvent,
+} from './preferences-controller';
+
+export type AppStateControllerState = {
+  timeoutMinutes: number;
+  connectedStatusPopoverHasBeenShown: boolean;
+  defaultHomeActiveTabName: string | null;
+  browserEnvironment: Record<string, string>;
+  popupGasPollTokens: string[];
+  notificationGasPollTokens: string[];
+  fullScreenGasPollTokens: string[];
+  recoveryPhraseReminderHasBeenShown: boolean;
+  recoveryPhraseReminderLastShown: number;
+  outdatedBrowserWarningLastShown: number | null;
+  nftsDetectionNoticeDismissed: boolean;
+  showTestnetMessageInDropdown: boolean;
+  showBetaHeader: boolean;
+  showPermissionsTour: boolean;
+  showNetworkBanner: boolean;
+  showAccountBanner: boolean;
+  trezorModel: string | null;
+  currentPopupId?: number;
+  onboardingDate: number | null;
+  lastViewedUserSurvey: number | null;
+  newPrivacyPolicyToastClickedOrClosed: boolean | null;
+  newPrivacyPolicyToastShownDate: number | null;
+  // This key is only used for checking if the user had set advancedGasFee
+  // prior to Migration 92.3 where we split out the setting to support
+  // multiple networks.
+  hadAdvancedGasFeesSetPriorToMigration92_3: boolean;
+  qrHardware: Json;
+  nftsDropdownState: Json;
+  usedNetworks: Record<string, boolean>;
+  surveyLinkLastClickedOrClosed: number | null;
+  signatureSecurityAlertResponses: Record<string, SecurityAlertResponse>;
+  // States used for displaying the changed network toast
+  switchedNetworkDetails: Record<string, string> | null;
+  switchedNetworkNeverShowMessage: boolean;
+  currentExtensionPopupId: number;
+  lastInteractedConfirmationInfo?: LastInteractedConfirmationInfo;
+  termsOfUseLastAgreed?: number;
+  snapsInstallPrivacyWarningShown?: boolean;
+  interactiveReplacementToken?: { url: string; oldRefreshToken: string };
+  noteToTraderMessage?: string;
+  custodianDeepLink?: { fromAddress: string; custodyId: string };
+};
+
+const controllerName = 'AppStateController';
+
+/**
+ * Returns the state of the {@link AppStateController}.
+ */
+export type AppStateControllerGetStateAction = {
+  type: 'AppStateController:getState';
+  handler: () => AppStateControllerState;
+};
```

Comment:
I don't think there is an handler registered for this event since this controller doesn't extend BaseController yet

Reply:
Actually @mikesposito is right, either remove the get state action or register an action handler at the controller level, but as this is not currently in use we can just remove it and can be easily added in BaseController upgrade. and event registration will be done automatically.